### DISCUSSION TITLE
Is there a built-in way to bundle assets?

### DISCUSSION THREAD

**Discussion by andrew-selvia** [ptrdom/scalajs-esbuild#203] [Q&A] [answered]
When I call `esbuildBundle` or `esbuildServe`, assets (e.g., css, png, svg) do not get bundled. I have tried placing them directly in the `esbuild` directory. I have also tried placing them within `esbuild/assets`.

I feel like this should this should work out-of-the-box; am I missing something?

**Comment by ptrdom** [comment, marked as answer]
You can see asset bundling in action in this example - https://github.com/ptrdom/scalajs-esbuild/tree/main/sbt-scalajs-esbuild-web/examples/basic-web-project. Caveat is that the assets have to be imported inside source files - for example, "loose" assets that are referenced only in `index.html` won't get put inside output directory. Only assets from `index.html` that are picked up are entry point files - `main.js` in the linked example case.

**Reply by andrew-selvia** [reply]
Ah... so, this:
```scala
@js.native
@JSImport("./javascript.svg", JSImport.Default)
object JavascriptLogo extends js.Object
```
will result in a reference in the compiled `main.js`, thereby triggering the asset to be copied.

That's good to know at least. I was stuck trying to get my favicon (referenced only in `index.html`) to be bundled. Sounds like I need to manually configure my sbt build to copy assets.

Thanks for the explanation @ptrdom!
